pytorch_lightning/trainer/progress.py

```diff
@@ -188,8 +191,8 @@ class BatchProgress(Progress):
 
     is_last_batch: bool = False
 
-    def reset_on_epoch(self) -> None:
-        super().reset_on_epoch()
+    def reset_on_run(self) -> None:
```

Comment:
let's add a docstring to clarify if the reset is before a run or when...
how about "reset_before_run"?

Reply:
It runs during `Loop.run` aka "on run"